apis/v1alpha1/route_types.go

```diff
@@ -64,6 +64,60 @@ type GatewayReference struct {
 	Namespace string `json:"namespace"`
 }
 
+// RouteForwardTo defines how a Route should forward a request.
+type RouteForwardTo struct {
+	// ServiceName refers to the name of the Service to forward matched requests
+	// to. When specified, this takes the place of BackendRef. If both
+	// BackendRef and ServiceName are specified, ServiceName will be given
+	// precedence. If the referent cannot be found, controllers must set the
+	// "InvalidRoutes" status condition on any Gateway that includes this
+	// Route to true.
+	//
+	// Support: Core
+	//
+	// +optional
+	// +kubebuilder:validation:MaxLength=253
+	ServiceName *string `json:"serviceName,omitempty"`
+
+	// BackendRef is a reference to a backend to forward matched requests to. If
+	// both BackendRef and ServiceName are specified, ServiceName will be given
+	// precedence. If the referent cannot be found, controllers must set the
+	// "InvalidRoutes" status condition on any Gateway that includes this Route
+	// to true.
+	//
+	// Support: Extended
+	//
+	// +optional
+	BackendRef LocalObjectReference `json:"backendRef,omitempty"`
+
+	// Port specifies the destination port number to use for the backend. If
+	// unspecified and a Service object consisting of a single port definition
+	// is the backend, that port will be used. If unspecified and the backend is
+	// a Service object consisting of multiple port definitions, controllers
+	// must set the "InvalidRoutes" status condition on any Gateway that
+	// includes this Route to true.
+	//
+	// Support: Core
+	//
+	// +optional
+	// +kubebuilder:validation:Minimum=1
```

Comment:
Is min needed for optional fields?

Reply:
I agree that we don't need to specify MinItems or MinLength for optional fields, but a Minimum is still important here so we ensure that if users do set this value it's not 0 or negative.